src/python_spy.rs

```diff
@@ -45,6 +45,14 @@ pub struct PythonSpy {
     pub dockerized: bool
 }
 
+fn exit_if_gil(config: &Config, version: &Version, msg: &str) {
+    if config.gil_only {
+        eprintln!("Cannot detect GIL holding in version '{}' on the current platform (reason: {})", version, msg);
+        eprintln!("Please open an issue in https://github.com/benfred/py-spy with the Python version and your platform.");
+        std::process::exit(1);
+    }
+}
```

Comment:
In the case where ```--gil``` wasn't being set, we used to produce a warning if it wasn't added. I think we should keep this warning when we're not exiting with an error:

```suggestion
    }
    warn!("Unable to detect GIL usage: {}", msg);
}
```

Reply:
Yeah I thought about it, but I realized the error is just redundant because if I didn't request `--gil` I probably don't care about seeing it. But ofc it's okay to add it back :)